runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3178,15 +3178,50 @@ gcReinitializeDefaultsForRestore(J9VMThread* vmThread)
 {
 	MM_GCExtensions* extensions = MM_GCExtensions::getExtensions(vmThread);
 	J9JavaVM *vm = vmThread->javaVM;
-	bool result = true;
 
 	PORT_ACCESS_FROM_JAVAVM(vm);
-
+	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+	/* Note here we update this parameter which represents the machine physical memory,
+	 * but not the original heap geometry from the snapshot run. We just force the heap
+	 * not to go beyond a certain thereshold calculated based on the new (restore) memory
+	 * availalibity through setting SoftMx.
+	 */
+	extensions->usablePhysicalMemory = omrsysinfo_get_addressable_physical_memory();
+	if (!extensions->userSpecifiedParameters._Xmx._wasSpecified){
+		uintptr_t candidateSoftMx = 0;
+		if (0.0 >= extensions->maxRAMPercent) {
+			/* We use false here for computeDefaultMaxHeapForJava(), since the restore
+			* path is only active for releases after Java 8 releases
+			*/
+			candidateSoftMx = extensions->computeDefaultMaxHeapForJava(false);
+		} else {
+			/* if the user have set maxRAMPercent through -XX:MaxRAMPercentage */
+			candidateSoftMx = extensions->maxRAMPercent * extensions->usablePhysicalMemory / 100.0;
+		}
+		/* we will set softMx value only if maxHeap calculation returned us a smaller
+		* value than existing maxHeap or softMx values and a larger value than existing
+		* minHeap value as inherited from/established at snapshot run, this max/minHeap
+		* is checked by j9gc_set_softmx()
+		*/
+		if (extensions->memoryMax > candidateSoftMx) {
+			if ((0 == extensions->softMx) || (extensions->softMx > candidateSoftMx)) {
+				if (extensions->initialMemorySize > candidateSoftMx) {
+					uintptr_t minimumSizeValue = extensions->initialMemorySize;
+					const char *qualifier = NULL;
+					qualifiedSize(&minimumSizeValue, &qualifier);
+					j9nls_printf(PORTLIB,J9NLS_ERROR,J9NLS_GC_SUBSPACE_TOO_SMALL_FOR_VALUE, "-Xsoftmx", minimumSizeValue, qualifier);
```

Comment:
I think this message can be very confusing for the case if "-Xsoftmx" has not been provided in the command line. I think this message should be improved.

Reply:
We can get rid of this with the current setup